servicex/app/init.py

```diff
@@ -0,0 +1,245 @@
+# Copyright (c) 2022, IRIS-HEP
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from pathlib import Path
+import getpass
+from typing import Literal, Optional, TypedDict
+import asyncio
+
+import typer
+from rich import get_console
+from rich.prompt import Prompt
+from rich.panel import Panel
+
+from servicex.servicex_adapter import ServiceXAdapter
+
+init_app = typer.Typer(name="init", invoke_without_command=True, no_args_is_help=False)
+
+
+class InitItem(TypedDict):
+    name: str
+    url: str
+    service_name: str
+
+
+class InitConfig(TypedDict):
+    uchicago: InitItem
+
+
+config: InitConfig = {
+    "uchicago": {
+        "name": "UChicago",
+        "url": "https://servicex.af.uchicago.edu",
+        "service_name": "servicex-uc-af",
+    },
+}
+
+
+async def verify_token(url: str, token: str) -> bool:
+    """Verify the token by pinging the ServiceX server."""
+    console = get_console()
+    try:
+        adapter = ServiceXAdapter(url=url, refresh_token=token)
+        await adapter.get_servicex_info()
+        return True
+    except Exception as e:
+        console.print(f"[red]✗ Failed to authenticate with ServiceX server:[/red] {e}")
+        return False
+
+
+def run(
+    source: Optional[Literal["uchicago"]] = None,
+    custom_url: Optional[str] = None,
+    auth_disabled: bool = False,
+):
+    if source is None and custom_url is None:
+        raise RuntimeError("An Access Facility source or specific url must be provided")
```

Comment:
Analysis Facility, presumably. Do you want to dump a list of pre-configured endpoints?